server/runtime/src/main/java/org/infinispan/server/Server.java

```diff
@@ -533,6 +546,21 @@ public void serializeConfiguration(ConfigurationWriter writer) {
       writer.writeEndDocument();
    }
 
+   private void startCacheManager(BlockingManager bm) {
+      bm.runBlocking(() -> {
```

Comment:
Can you explain why this is needed? Shouldn't this be the main thread doing this work in the first place?